README.md

```diff
@@ -126,7 +136,7 @@ The important features are:
 The most basic run command for this pipeline is: 
 
 ```
-nextflow run main.nf --input samples.csv --ref reference.fasta  --intervalList_path path_to_intervals --ponvcf pon.vcf.gz
+nextflow run main.nf --input samples.csv --ref reference.fasta --ponvcf pon.vcf.gz
```

Comment:
I have tried to identify the mandatory parameters and a couple of optional ones.

I have a question which I am unable to get my head around. Can you suggest what could be the problem?

In the run_pipeline_on_gadi_script.sh script , if
-  I remove any of the mandatory parameters from the command , e.g. **--ref ${ref}** from the command below , the if condition in main.nf catches it and displays the help dialog and prevents execution of the pipeline.

nextflow run main.nf \
        --input ${samples} \
        -profile gadi \
        --whoami $(whoami) --gadi_account $PROJECT \
         --dict ${dict} \
	--common_biallelic_variants ${common_biallelic_variants} \
        --ponvcf ${ponvcf} \
        --outDir ${outDir} \
        --number_of_intervals ${number_of_intervals}

However if 
- I inactivate (using #) the parameter
ref=/g/data/er01/SIH-HPC-WGS/Reference/hs38DH.fasta
 but retain  it in the command as below
nextflow run main.nf \
        --input ${samples} \
        -profile gadi \
        --whoami $(whoami) --gadi_account $PROJECT \
        **--ref ${ref}** --dict ${dict} \
	--common_biallelic_variants ${common_biallelic_variants} \
        --ponvcf ${ponvcf} \
        --outDir ${outDir} \
        --number_of_intervals ${number_of_intervals}
        
The value displayed for ref = true and I get an error!

I am concerned that the user can do one of the above (by mistake) and both the options should lead to identical help display message!

Thanks.

Reply:
Bit confused by this, why would anyone hash the `ref=` variable in the bash script if it is a required argument and your instructions document that?   

You've used an if statement to ensure the pipeline doesn't run unless all those required params are provided: https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/0e9f22d1b3bc1b4ff7b43aa1fc5db78afba41ed9/main.nf#L83

The `ref` parameter is defined as false in `nextflow.config`: https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/0e9f22d1b3bc1b4ff7b43aa1fc5db78afba41ed9/nextflow.config#L18

But you're still passing an (albeit empty) value via your run command when using `--ref ${ref}`. To nextflow it might look like `--ref <space>` so that might be why its displaying as `true`.

Reply:
> Bit confused by this, why would anyone hash the `ref=` variable in the bash script if it is a required argument and your instructions document that?

**By mistake I guess**; So I was thinking of stopping the user .. But I agree. `ref=` variable is a mandatory requirement.



> You've used an if statement to ensure the pipeline doesn't run unless all those required params are provided:
> 
> https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/0e9f22d1b3bc1b4ff7b43aa1fc5db78afba41ed9/main.nf#L83

BTW The above if condition is not a part of the update branch main.nf at https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/autogen_intervals_issue5/main.nf
This is how it looks like now: 
  if ( params.help == true || params.ref == false || params.input == false || params.ponvcf == false || params.common_biallelic_variants == false || params.dict == false)


> 
> The `ref` parameter is defined as false in `nextflow.config`:
> 
> https://github.com/Sydney-Informatics-Hub/Somatic-shortV-nf/blob/0e9f22d1b3bc1b4ff7b43aa1fc5db78afba41ed9/nextflow.config#L18
> 
> But you're still passing an (albeit empty) value via your run command when using `--ref ${ref}`. To nextflow it might look like `--ref <space>` so that might be why its displaying as `true`.
**OK Thanks.**

Reply:
What do you mean stopping the user? How would you do this?